common/plugins/eu.esdihumboldt.hale.common.instance.graph/src/eu/esdihumboldt/hale/common/instance/graph/reference/ReferenceGraph.java

```diff
@@ -235,24 +241,24 @@ public Boolean compute(LoopBundle<Vertex> loop) {
 					if (ref != null) {
 						result.add(ref);
 					}
-					else {
+					else if (!vtx.getId().equals(associated.getId())) {
```

Comment:
So there were references from one instance to itself?

Reply:
Yes and that was the case when actually the warning was given.

Reply:
Did you identify how that can happen? I don't remember self-references from the test data that we looked at and even if there were I would expect the `ref != null` condition to be `true`.
Any idea? Just wondering if something is not correct when building the graph.

Reply:
Yes, the condition `vtx.getId().equals(associated.getId())` is `true` for the example attached to the issue many times, as well as for the sample example that we have created!